community/kernel/src/main/java/org/neo4j/kernel/impl/locking/community/RWLock.java

```diff
@@ -146,42 +172,53 @@ synchronized void acquireReadLock( Object tx ) throws DeadlockDetectedException
     {
         TxLockElement tle = getOrCreateLockElement( tx );
 
+        LockRequest lockRequest = null;
+        boolean addLockRequest = true;
```

Comment:
Concept is still the same as with `WaitElement` except it gives more specific description what we are requesting, also completion flag introduced to be able to mark particular request as completed to avoid duplicated registration of lock that already was registered before.